cpp/src/arrow/dataset/CMakeLists.txt

```diff
@@ -207,8 +207,8 @@ if(ARROW_PARQUET)
 endif()
 
 if(ARROW_BUILD_BENCHMARKS)
-  add_arrow_benchmark(file_benchmark PREFIX "arrow-dataset")
-  add_arrow_benchmark(scanner_benchmark PREFIX "arrow-dataset")
+  add_arrow_compute_benchmark(file_benchmark PREFIX "arrow-dataset")
+  add_arrow_compute_benchmark(scanner_benchmark PREFIX "arrow-dataset")
```

Comment:
Feels weird of adding a dataset benchmark as "compute benchmark".

Reply:
should a name with something like `add_arrow_register_kernels_benchmark` or something like this work better?
I could create a function just for dataset `add_arrow_dataset_benchmark` but that would just be a wrapper

Reply:
I think having a wrapper is nice. It not only hides the underlying implementation (leveraging `add_arrow_compute_benchmark`) thus making the invocation more explicit, but also sorts out other subproject specific logic (such as prepending the name prefix).